File: 3159-find-occurrences-of-an-element-in-an-array/3159-find-occurrences-of-an-element-in-an-array.rb
# @param {Integer[]} nums
# @param {Integer[]} queries
# @param {Integer} x
# @return {Integer[]}
def occurrences_of_element(nums, queries, x)
  occurrences = []
  nums.each_with_index { |num, i| occurrences << i if num == x }

  queries.map do |query|
    if query <= occurrences.size
      occurrences[query - 1]
    else
      -1
    end
  end
  
end

File: 3159-find-occurrences-of-an-element-in-an-array/3159-find-occurrences-of-an-element-in-an-array.js
/**
 * @param {number[]} nums
 * @param {number[]} queries
 * @param {number} x
 * @return {number[]}
 */
var occurrencesOfElement = function(nums, queries, x) {
    let occurrences = [];
    nums.forEach((num, i) => {
        if (num === x) occurrences.push(i);
    });

    return queries.map(query => {
        return query <= occurrences.length ? occurrences[query - 1] : -1;
    });
};